OpenRA.Mods.Common/Widgets/Logic/Ingame/DebugLogic.cs

```diff
@@ -0,0 +1,45 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Network;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic.Ingame
+{
+	public class DebugLogic : ChromeLogic
+	{
+		[ObjectCreator.UseCtor]
+		public DebugLogic(Widget widget, OrderManager orderManager, World world, WorldRenderer worldRenderer)
+		{
+			var geometryOverlay = world.WorldActor.TraitOrDefault<TerrainGeometryOverlay>();
+			if (geometryOverlay != null)
+			{
+				var labelWidget = widget.Get<LabelWidget>("DEBUG_TEXT");
+
+				labelWidget.GetText = () =>
+				{
+					var cell = worldRenderer.Viewport.ViewToWorld(Viewport.LastMousePos);
+					var map = worldRenderer.World.Map;
+
+					var worldPx = worldRenderer.Viewport.ViewToWorldPx(Viewport.LastMousePos);
+					var worldPos = worldRenderer.ProjectedPosition(worldPx);
+
+					return map.Height.Contains(cell) ?
+						"{0}, ({1})".F(cell, worldPos) : "";
```

Comment:
IMO the comma after the Y coordinate looks weird, can we use "{0} ({1})" instead?